hr_attendance_reason/static/src/js/attendance_reason.js

```diff
@@ -0,0 +1,69 @@
+odoo.define("hr_attendance_reason.attendance_reason", function(require) {
+    "use strict";
+
+    var MyAttendances = require("hr_attendance.my_attendances");
+    var core = require("web.core");
+    var _t = core._t;
+
+    MyAttendances.include({
+        // eslint-disable-next-line no-unused-vars
+        init: function(parent, action) {
+            var self = this;
+            this._super.apply(this, arguments);
+            this._rpc({
+                model: "res.company",
+                method: "search_read",
+                args: [
+                    [["id", "=", this.getSession().company_id]],
+                    ["attendance_reasons_in_kiosk"],
+                ],
+            }).then(function(res) {
+                self.attendance_reasons_in_kiosk = res[0].attendance_reasons_in_kiosk;
+            });
+            this.reasons = this.getAttendanceReasons();
+        },
+        getAttendanceReasons: function() {
+            var self = this;
+            this.reasons = [];
+            this._rpc({
+                model: "hr.attendance.reason",
+                method: "search_read",
+                fields: ["name", "action_type"],
+                domain: [["show_in_kiosk_mode", "=", true]],
+            }).then(function(reasons) {
+                _.each(reasons, function(reason) {
+                    self.reasons.push(reason);
+                });
+            });
+            return this.reasons;
+        },
+        update_attendance: function() {
```

Comment:
Isn't this shadowing the original function?

Reply:
Yes, the whole function is replaced. Reviewing the code of `hr_attendance` I see that it is a "hell" inheritance (not possible) and it is necessary to re-define some methods completely again. Any suggestions on how to simplify the code (js especially)?

Reply:
Not easy to solve and we'll clash with this other module which does the same monkeypatching https://github.com/OCA/hr-attendance/blob/cc0234df35d7fda1d090c121f0799d5c7fc70e72/hr_attendance_geolocation/static/src/js/attendance_geolocation.js#L14-L28

Reply:
For this case I researched a solution that fits the case. For the other module, I should find out.

Reply:
OK, let's use that method then for now.